src/__tests__/posthog-core.js

```diff
@@ -36,7 +36,9 @@ describe('identify()', () => {
         expect(given.overrides.register).toHaveBeenCalledWith({ distinct_id: 'a-new-id' })
     })
 
-    it('calls capture when identity changes', () => {
+    it('calls capture when identity changes and old ID is anonymous', () => {
+        given.lib.persistence = { $device_id: 'oldIdentity' }
```

Comment:
Nit: The test is not consistent with how given2 is normally used, causing unneeded repetition between tests.

The way to do this via given2 is following.

Above:
```js
given('overrides', () => ({
  persistence: { $device_id: given.deviceId }
  // same as rest

given('deviceId', () => 'oldIdentity')

// Drop lines like given.lib.persistence = in both tests
// In the second test
given('deviceId', () => 'anonymousId')
```

Reply:
I spent ~30 mins trying to figure out how to do this (in combo with using get_property like above) and couldn't work it out, so i've gone down the previous route again.

Reply:
Yikes! Sorry about that - happy to help set things up better next time.

Until then, pushed a quick fix myself.